frame/democracy/src/lib.rs

```diff
@@ -525,35 +559,58 @@ decl_module! {
 
 		fn deposit_event() = default;
 
+		fn on_runtime_upgrade() -> Weight {
+			if StorageVersion::get() == Releases::V1 {
+				DepositOf::<T>::translate::<
+					(BalanceOf<T>, Vec<T::AccountId>), _
+				>(|_, (balance, accounts)| {
+					Some((accounts, balance))
+				});
+
+				StorageVersion::put(Releases::V2);
+
+				T::MaximumBlockWeight::get()
```

Comment:
Why return the maximum block weight and not a closer approximation?

Reply:
I thought it doesn't harm to make a runtime_upgrade take the whole block weight. but we can be more precise indeed

Reply:
full block sounds fine to me